### ISSUE TITLE
Confusing error message when canceling "Start Interactive Rebase"

### ISSUE THREAD

**Issue by Lin-Buo-Ren** [git-cola/git-cola#826]
## Issue Reproducing Instructions
1. Launch Git Cola to open an existing repository
1. Actions > Rebase > Start Interactive Rebase
1. Input `HEAD~1` in the "Select New Upstream" prompt and click "Interactive Rebase" button
1. Close the "Rebase onto HEAD~1` window by pressing the X button on the title bar

## Expected Behavior
The window closed without error, the rebase completed with default order just like running `git rebase --interactive HEAD~1` in terminal and exit the editor without saving content.

## Current Behavior
An "Rebased stopped" error dialog popped with `"git rebase" returned exit status 1` as content and `Could not execute editor` in "Show Details"

### Git Cola
3.1 (v3.1-28-gd5caab1b) 

- Linux-4.13.0-41-generic-x86_64-with-neon-16.04-xenial 
- Python (/usr/bin/python) 2.7.12 (default, Dec 4 2017, 14:50:18) [GCC 5.4.0 20160609] 
- Git 2.17.0 
- Qt 5.10.0 
- QtPy 1.3.1 
- PyQt5 5.9.2

**Comment by davvid** [comment]
Yeah, unfortunately the `git rebase` tooling doesn't allow us to distinguish between exits from the editor and actual errors.  We have to return a non-zero exit code in order to cancel the rebase.

I think we might have a workaround -- instead of returning a non-zero exit code, we can empty out the file.  This is kinda against the "safe" approach, which is to not touch the instruction sheet file at all, but worth doing because it would result in better UX.

What we can do is not touch that file by default, and set an environment variable inside cola that triggers this behavior.  That way if someone is using the `git-xbase` editor standalone outside of cola we won't be suddenly changing to a non-destructive behavior on them.  By someone I also mean myself, since the way I test that tool is to run it on an instruction sheet standalone outside of cola.

I think that should resolve this issue -- I'll give that a try.
